python/ray/tune/analysis/experiment_analysis.py

```diff
@@ -119,6 +123,49 @@ def get_all_configs(self, prefix=False):
                 "Couldn't read config from {} paths".format(fail_count))
         return self._configs
 
+    def get_trial_checkpoints_paths(self, trial, metric=TRAINING_ITERATION):
+        """Returns a list of [path, metric] lists for all disk checkpoints of
+         a trial.
+
+        Arguments:
+            trial(Trial): The log directory of a trial, or a trial instance.
+            metric (str): key for trial info to return, e.g. "mean_accuracy".
+            "training_iteration" is used by default.
+        """
+
+        if isinstance(trial, str):
+            trial_dir = os.path.expanduser(trial)
+            if not os.path.isdir(trial_dir):
+                raise ValueError(
+                    "{} is not a valid directory.".format(trial_dir))
+
+            # get checkpoints from logdir
```

Comment:
Let me know if there is a better way to find all the checkpoints in a logdir. Thanks.

Reply:
I think there's a marker file in each checkpoint_dir now

Reply:
maybe use that instead? @ujvl also introduced another method for determining checkpoint paths too in checkpointmanager?

Reply:
Actually, the method only looks up the tree...

Reply:
See TrainableUtil.make_checkpoint_dir

Reply:
Updated to use it. Thanks.